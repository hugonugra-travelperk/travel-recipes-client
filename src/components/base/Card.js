import styled from "styled-components";

export const CardWrapper = styled.div`
  overflow: hidden;
  padding: 0 0 3em;
  margin: 2em;
  font-family: Quicksand, arial, sans-serif;
  border-radius: 0.25rem;
  box-shadow: 0 3em 4em -1em rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  overflow: hidden;
`;

export const CardHeader = styled.header`
  padding-top: 0.2em;
  padding-bottom: 0.2em;
  font-size: 1.75em;
  font-weight: bold;
  text-align: center;
`;

export const CardBody = styled.div`
  padding-right: 1em;
  padding-left: 1em;
  padding-top: 0.75em;
  font-size: 1em;
  margin: 2em;
  border: 0.1em solid blue;
`;

export const CardSubHeader = styled.header`
  padding-top: 0.2em;
  padding-bottom: 0.2em;
  font-size: 1.25em;
  font-weight: bold;
  text-align: center;
`;

export const CardSubBody = styled.div`
  padding: 1em;
  font-size: 1em;
  text-align: center;
  background: blue;
  margin-top: 0.8em;
  border-radius: 0.5rem;
`;
