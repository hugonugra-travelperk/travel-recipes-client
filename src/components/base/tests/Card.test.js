import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import {CardWrapper, CardHeader, CardBody, CardSubHeader, CardSubBody} from "../Card"

afterEach(cleanup);

it("renders", () => {
  const {asFragment} = render(<CardWrapper>Test text</CardWrapper>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const {asFragment} = render(<CardHeader>Test text</CardHeader>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const {asFragment} = render(<CardBody>Test text</CardBody>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const {asFragment} = render(<CardSubHeader>Test text</CardSubHeader>);
  expect(asFragment()).toMatchSnapshot();
});

it("renders", () => {
  const {asFragment} = render(<CardSubBody>Test text</CardSubBody>);
  expect(asFragment()).toMatchSnapshot();
});
