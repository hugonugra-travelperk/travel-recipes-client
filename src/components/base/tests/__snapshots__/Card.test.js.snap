// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL gaBRgU"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`renders 2`] = `
<DocumentFragment>
  <header
    class="sc-gsDKAQ jGoJmq"
  >
    Test text
  </header>
</DocumentFragment>
`;

exports[`renders 3`] = `
<DocumentFragment>
  <div
    class="sc-dkPtRN PHDyg"
  >
    Test text
  </div>
</DocumentFragment>
`;

exports[`renders 4`] = `
<DocumentFragment>
  <header
    class="sc-hKwDye klnJXo"
  >
    Test text
  </header>
</DocumentFragment>
`;

exports[`renders 5`] = `
<DocumentFragment>
  <div
    class="sc-eCImPb gayBGf"
  >
    Test text
  </div>
</DocumentFragment>
`;
